x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/action/token/TransportInvalidateTokenActionTests.java

```diff
@@ -0,0 +1,119 @@
+import org.elasticsearch.common.io.stream.OutputStreamStreamOutput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.license.XPackLicenseState;
+import org.elasticsearch.node.Node;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.test.ClusterServiceUtils;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.threadpool.TestThreadPool;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+import org.elasticsearch.xpack.core.XPackSettings;
+import org.elasticsearch.xpack.core.security.SecurityContext;
+import org.elasticsearch.xpack.core.security.action.token.InvalidateTokenRequest;
+import org.elasticsearch.xpack.core.security.action.token.InvalidateTokenResponse;
+import org.elasticsearch.xpack.security.authc.AuthenticationService;
+import org.elasticsearch.xpack.security.authc.TokenService;
+import org.elasticsearch.xpack.security.support.SecurityIndexManager;
+import org.junit.After;
+import org.junit.Before;
+
+import java.io.ByteArrayOutputStream;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.time.Clock;
+import java.util.Base64;
+import java.util.Collections;
+
+import static org.elasticsearch.xpack.core.security.action.token.InvalidateTokenRequest.Type.ACCESS_TOKEN;
+import static org.elasticsearch.xpack.core.security.action.token.InvalidateTokenRequest.Type.REFRESH_TOKEN;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class TransportInvalidateTokenActionTests extends ESTestCase {
+
+    private static final Settings SETTINGS = Settings.builder().put(Node.NODE_NAME_SETTING.getKey(), "TokenServiceTests")
+        .put(XPackSettings.TOKEN_SERVICE_ENABLED_SETTING.getKey(), true).build();
+
+    private ThreadPool threadPool;
+    private Client client;
+    private SecurityIndexManager securityIndex;
+    private ClusterService clusterService;
+    private AuthenticationService authenticationService;
+    private XPackLicenseState license;
+    private SecurityContext securityContext;
+
+    @Before
+    public void setup() {
+        threadPool = new TestThreadPool(getTestName());
+        securityContext = new SecurityContext(Settings.EMPTY, threadPool.getThreadContext());
+        client = mock(Client.class);
+        when(client.threadPool()).thenReturn(threadPool);
+        when(client.settings()).thenReturn(SETTINGS);
+        securityIndex = mock(SecurityIndexManager.class);
+        this.clusterService = ClusterServiceUtils.createClusterService(threadPool);
+        this.license = mock(XPackLicenseState.class);
+        when(license.isTokenServiceAllowed()).thenReturn(true);
+    }
+
+    public void testInvalidateTokensWhenIndexUnavailable() throws Exception {
```

Comment:
This was added because we couldn't trigger the index to be unavailable ( but rather only not existent ) in TokenAuthIntegTests

Reply:
Is it worthwhile to test "unavailable" index in `TokenAuthIntegTests` by closing the token index before invalidation?